forge-game/src/main/java/forge/game/ability/effects/ExploreEffect.java

```diff
@@ -45,54 +47,60 @@ protected String getStackDescription(SpellAbility sa) {
      */
     @Override
     public void resolve(SpellAbility sa) {
-        final Game game = sa.getHostCard().getGame();
+        final Card host = sa.getHostCard();
+        final Game game = host.getGame();
+        int amount = AbilityUtils.calculateAmount(host, sa.getParamOrDefault("Num", "1"), sa);
 
         GameEntityCounterTable table = new GameEntityCounterTable();
         final CardZoneTable triggerList = new CardZoneTable();
         Map<AbilityKey, Object> moveParams = AbilityKey.newMap();
         moveParams.put(AbilityKey.LastStateBattlefield, sa.getLastStateBattlefield());
         moveParams.put(AbilityKey.LastStateGraveyard, sa.getLastStateGraveyard());
         for (final Card c : getTargetCards(sa)) {
```

Comment:
The changeZone and Counter Trigger needs to be inside the For Amount Loop

so "Once" Replacement/Trigger would trigger multiple times

Reply:
Is this one covered, or does it need more work?